src/main/php/PHPMD/Rule/UnusedFormalParameter.php

```diff
@@ -95,8 +95,8 @@ protected function isAbstractMethod(AbstractNode $node)
      */
     protected function isInheritedSignature(AbstractNode $node)
     {
-        if ($node instanceof MethodNode) {
-            return preg_match('/@inheritdoc/i', $node->getDocComment()) === 1;
+        if ($node instanceof MethodNode && ($comment = $node->getDocComment()) !== null) {
+            return preg_match('/@inheritdoc/i', $comment) === 1;
```

Comment:
```suggestion
        if ($node instanceof MethodNode) {
            $comment = $node->getDocComment();

            return $comment && preg_match('/@inheritdoc/i', $comment);
```
We promote not to put assignments inside conditions (https://phpmd.org/rules/cleancode.html#ifstatementassignment) so maybe the above may be a cleaner option and a way to follow our own recommendations.

Reply:
@AJenbo is it the part of the code-style you were mentioning?

Reply:
The code you suggested looks better. :)
https://github.com/phpmd/phpmd/pull/928/commits/a4f9f2df5ba71a17b630a945a65cc0c3d7eea793